### ISSUE TITLE
Inconsistent ONNX Export with Differentiable Head

### ISSUE THREAD

**Issue by radandreicristian** [huggingface/setfit#418]
Hello,

I am trying to fine-tune SetFit for a multi-class classification problem.

Everything is smooth until exporting to Onnx. The head is not exported correctly, so when loading the model with Onnx, the predictions are the outputs of some previous layer.

For example, I showcased this on a dummy dataset with 3 classes.

See https://colab.research.google.com/drive/19EESqbIDwD5FOI2Ufx22txFZ8qADq2xj?authuser=1#scrollTo=vl6AvQKqtU-9

This differs from the behaviour in the example notebook, where the LogisticRegression head is used.

Any directions would be appreciated; Otherwise, I would happily contribute with an MR if anyone can spot the issue.

**Comment by radandreicristian** [comment]
The output is the raw logits per class for each sample, and all you need to do then is a softmax/argmax to get the class labels.

Is this intended?

**Comment by tomaarsen** [comment]
Hello!

I'm aware that there's some issues with exporting ONNX sadly. Hopefully I will have some time in the future to refactor the exporting to a more consistent approach instead. Thank you for raising this issue & for providing a Google Colab! It'll be helpful for certain.

- Tom Aarsen

**Comment by pedrogengo** [comment]
Hey @tomaarsen, can I work on this?

**Comment by pedrogengo** [comment]
https://colab.research.google.com/drive/19xE4WdxqGLLZOSanycYfUzbcAxFgpzuR?usp=sharing

Here is my colab that I used to debug
